File: src/main/java/com/github/asufana/presentationjava/PresentationImpl.java
package com.github.asufana.presentationjava;

import com.github.asufana.presentationjava.fields.Fields;
import com.github.asufana.presentationjava.fields.FieldsFactory;

public class PresentationImpl {

    /** HTML変換（デフォルトレイアウト） */
    public static String toHtml(final Object targetInstance) {

        //フィールド一覧取得
        final Fields fields = FieldsFactory.create(targetInstance);

        //詳細画面向けHTML文字列生成
        final String html = fields.toDescTableHtml();

        return html;
    }

    /** HTML変換 */
    public static String toHtml(final Object targetInstance, String layout) {

        //フィールド一覧取得
        final Fields fields = FieldsFactory.create(targetInstance);

        //詳細画面向けHTML文字列生成
        final String html = fields.toDescTableHtml(layout);

        return html;
    }

}

File: src/main/java/com/github/asufana/presentation/PresentationImpl.kt
package com.github.asufana.presentation

import com.github.asufana.presentation.fields.FieldsFactory

class PresentationImpl {

    companion object {

        /** HTML変換（デフォルトレイアウト） */
        @JvmStatic
        fun toHtml(targetInstance: Any): String {

            //フィールド一覧
            val fields = FieldsFactory.create(targetInstance)

            //HTML生成
            return fields.toDescTableHtml()
        }

        /** HTML変換 */
        @JvmStatic
        fun toHtml(targetInstance: Any, layout: String): String {

            //フィールド一覧
            val fields = FieldsFactory.create(targetInstance)

            //HTML生成
            return fields.toDescTableHtml(layout)
        }

    }

}


